src/Microsoft.Security.Utilities.Benchmarks/SecretMaskerDetectionBenchmarks.cs

```diff
@@ -5,42 +5,85 @@
-        // hit less concentrated in the profiling.
-        private const int SecretPrefixSize = 100 * 1024;
+        DotNet,
+        RE2,
+    }
+
+    public enum Case
+    {
+        // This is close to the worst case scenario for large input. Every
+        // base64 alphabet char is equally likely to appear in the input, making
+        // signature sniffing harder.
+        HardPrefix,
+
+        // This is close to the best case scenario. A char that does not appear
+        // in any identifiable key signature is repeated in the prefix, making
+        // signature sniffing easier.
+        EasyPrefix,
+
+        // This will scan secrets without prefixing them. Testing the performance against
+        // small inputs.
+        NoPrefix,
+    }
 
-        protected List<string> RandomExamples { get; }
-        protected List<string> NonRandomExamples { get; }
+    public abstract class SecretMaskerDetectionBenchmarks
+    {
+        // The size of randomized data to add as a prefix for every secret for
+        // the EasyPrefix and HardPrefix cases. This is intended to make
+        // positive hits less concentrated in the profiling.
+        public const int SecretPrefixSize = 10 * 1024;
+
+        // Whether to use the high performance scanner. Normally, the high
+        // performance scanner is always used, but there's a test hook to turn
+        // it off, which we use to see how much it is optimizing.
+        [Params(true, false)]
+        public bool UseHighPerformanceScanner { get; set; } = true;
+
+        // Whether to generate correlating ids for each match. Setting this to
+        // true will contribute fixed hash production overhead to all the
+        // scanners. Uncomment 'Params' attribute to benchmark with and without
+        // it.
+        //
+        //[Params(true,  false)]
+        public bool GenerateCorrelatingIds { get; set; } = false;
+
+        // The regex engine to use. The DotNet engine is faster than RE2 on
+        // modern .NET, and RE2 is faster than DotNet on .NET Framework for
+        // large input. Uncomment Params attribute to benchmark both engines,
+        // Otherwise, we'll use the better default for the current platform.
+        //
+        //[Params(RegexEngine.DotNet, RegexEngine.RE2)]
+        public RegexEngine RegexEngine { get; set; } =
+#if NET
+            RegexEngine.DotNet;
+#else
+            RegexEngine.RE2;
+#endif
+
+        [ParamsAllValues]
+        public Case Case { get; set; } = Case.HardPrefix;
 
         // Use a fixed seed to ensure different runs use the same data.
-        private readonly Random rng = new Random(Seed: 42);
+        private readonly Random _rng = new Random(Seed: 42);
 
-        protected SecretMaskerDetectionBenchmarks()
-        {
-            RandomExamples = GenerateExamples(random: true);
-            NonRandomExamples = GenerateExamples(random: false);
-        }
+        protected List<string> Examples { get; private set; } = null!;
```

Comment:
>List [](http://example.com/codeflow?start=18&length=4)

SEV4 perhaps `IList`? is it possible someone might want to provide a sorted list for example?

Reply:
I'd prefer to leave this. The only use case is in a derived benchmark that wants to bench scanning the examples with a different scanner. (Which is done in the internal benchmark). In that case, I want enumeration of the examples to take as little of the benchmark time as possible.